contracts/KeyGenHistory.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity 0.5.9;
+pragma experimental ABIEncoderV2;
+
+
+contract KeyGenHistory {
+
+    mapping(address => bytes) public parts;
+    mapping(address => bytes) public acks;
```

Comment:
We need to allow multiple `Ack`s per validator.
If there are _N_ validators, each of them is supposed to send _1_ `Part` and _N_ `Ack` messages.